client/app/components/CreateSourceDialog.jsx

```diff
@@ -0,0 +1,191 @@
+
+class CreateSourceDialog extends React.Component {
+  static propTypes = {
+    dialog: DialogPropType.isRequired,
+    types: PropTypes.arrayOf(PropTypes.object),
+    sourceType: PropTypes.string.isRequired,
+    imageFolder: PropTypes.string.isRequired,
+    helpTriggerPrefix: PropTypes.string,
+    onCreate: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    types: [],
+    helpTriggerPrefix: null,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchText: '',
+      selectedType: null,
+      savingSource: false,
+      currentStep: StepEnum.SELECT_TYPE,
+    };
+  }
+
+  selectType = (selectedType) => {
+    this.setState({ selectedType, currentStep: StepEnum.CONFIGURE_IT });
+  };
+
+  resetType = () => {
+    if (this.state.currentStep === StepEnum.CONFIGURE_IT) {
+      this.setState({ searchText: '', selectedType: null, currentStep: StepEnum.SELECT_TYPE });
+    }
+  };
+
+  createSource = (values, successCallback, errorCallback) => {
+    const { selectedType, savingSource } = this.state;
+    if (!savingSource) {
+      this.setState({ savingSource: true, currentStep: StepEnum.DONE });
+      this.props.onCreate(selectedType, values).then(() => {
+        successCallback('Saved.');
+        this.props.dialog.close({ success: true });
+      }).catch((error) => {
+        this.setState({ savingSource: false, currentStep: StepEnum.CONFIGURE_IT });
+        errorCallback(error.message);
+      });
+    }
+  };
+
+  renderTypeSelector() {
+    const { types } = this.props;
+    const { searchText } = this.state;
+    const filteredTypes = types.filter(type => isEmpty(searchText) ||
+      includes(type.name.toLowerCase(), searchText.toLowerCase()));
+    return (
+      <div className="m-t-10">
+        <Search
+          placeholder="Search..."
+          onChange={e => this.setState({ searchText: e.target.value })}
+          autoFocus
+          data-test="SearchSource"
+        />
+        <div className="scrollbox p-5 m-t-10" style={{ minHeight: '30vh', maxHeight: '40vh' }}>
+          {isEmpty(filteredTypes) ? (<EmptyState className="" />) : (
+            <List
+              size="small"
+              dataSource={filteredTypes}
+              renderItem={item => this.renderItem(item)}
+            />
+          )}
+        </div>
+      </div>
+    );
+  }
+
+  renderForm() {
+    const { types, imageFolder, helpTriggerPrefix } = this.props;
+    const { selectedType } = this.state;
+    const type = find(types, { type: selectedType });
+    const fields = helper.getFields(type.configuration_schema);
+    const helpTriggerType = `${helpTriggerPrefix}${toUpper(selectedType)}`;
+    return (
+      <div className="p-5">
+        <div className="d-flex justify-content-center align-items-center">
+          <img
+            className="profile__image--settings"
+            src={`${imageFolder}/${selectedType}.png`}
+            alt={type.name}
+            width="48"
+          />
```

Comment:
This one was tricky because ElasticSearch logo is an exception, all the other images already have some spacing. I tested with 5 px of padding in the image here, it looked acceptable to me.

Reply:
Since the ES logo needs updating anyway (it's an orange square now) we can just make sure it's got spacing like all the other kids.